src/main/java/org/yaml/snakeyaml/scanner/ScannerImpl.java

```diff
@@ -1893,9 +1893,7 @@ private Token scanFlowScalar(char style) {
   /**
    * Scan some number of flow-scalar non-space characters.
    */
-  private String scanFlowScalarNonSpaces(boolean doubleQuoted, Mark startMark) {
-    // See the specification for details.
-    StringBuilder chunks = new StringBuilder();
```

Comment:
Each invocation was creating a `new StringBuilder`, then constructing a new `String` result, and appending to the existing `StringBuilder` in `scanFlowScalar`. Since these are private, we can pass the one `StringBuilder` through append directly to it.

![image](https://github.com/user-attachments/assets/eb5cf4cb-e166-4586-ab1c-9d838bbe897a)